src/app/shared/components/template/services/template-metadata.service.ts

```diff
@@ -13,64 +14,32 @@ import { Capacitor } from "@capacitor/core";
 @Injectable({
   providedIn: "root",
 })
-export class TemplateMetadataService extends SyncServiceBase implements OnDestroy {
-  public parameterList = signal<FlowTypes.Template["parameter_list"]>({});
-  private enabled: boolean;
-  private destroy$ = new Subject();
+export class TemplateMetadataService extends SyncServiceBase {
+  /** Utility snapshot used to get router snapshot from service (outside render context) */
+  private snapshot = toSignal(ngRouterMergedSnapshot$(this.router));
+
+  /** Name of current template provide by route param */
+  private templateName = computed<string | undefined>(() => this.snapshot().params.templateName);
+
+  /** List of parameterList provided with current template */
+  public parameterList = signal<FlowTypes.Template["parameter_list"]>({}, { equal: isEqual });
 
   constructor(
     private templateService: TemplateService,
     private router: Router
   ) {
     super("TemplateMetadata");
 
-    // Currently the only watched parameter is for screen orientation,
-    // which is only supported on native platforms
-    this.enabled = !Capacitor.isNativePlatform();
```

Comment:
I can see how setting the enabled state based on a child dependency isn't sensible. But I'm aware that this service adds logic that runs on every route change and will only have an effect on one deployment currently, which makes use of the screen orientation feature. Do you think it would be sensible to do something like run a check of all templates on init to see whether any make use of the `parameter_list` properties tracked by this service, and only enable it in the case that such a template exists?

Either way, I imagine the performance impact of this service is minimal, so I'm happy to merge as-is.

Reply:
Yeah as this has no UI re-rendering effect and that templates change at most once per few seconds I can't see this really doing much to impact performance, so would make sense to optimise as you've suggested (from the templating). We're already planning to do this for other modules/features, so that if not in use the entire service will be excluded from the build (no need to toggle in the service itself). This should hopefully keep the code a bit simpler.

Template analysis won't know anything about native/web devices, so good to keep those checks in other services